presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java

```diff
@@ -230,6 +240,25 @@ public void testArrays()
         assertFilterProjectFails("nested_keys[2][5] > 0", "orderkey", "Array subscript out of bounds");
     }
 
+    @Test
+    public void testStructs()
+    {
+        assertQueryUsingH2Cte("SELECT orderkey, info, dates FROM lineitem_ex");
+
+        assertQuery("SELECT info.orderkey FROM lineitem_ex", WITH_LINEITEM_EX + " SELECT info[0] FROM lineitem_ex");
```

Comment:
nit : add a helper method makes it easier

```
assertQueryUsingH2Cte("SELECT info.orderkey FROM lineitem_ex", " SELECT info[0] FROM lineitem_ex");

```
```
private void assertQueryUsingH2Cte(String query, String h2query) {
        assertQuery(query, WITH_LINEITEM_EX + toH2(h2query));
    }

```

Reply:
@bhhari Indeed, it would be nice to not have to type the query twice. However, H2 doesn't support structs, hence, `SELECT info.orderkey FROM lineitem_ex ...` needs to be manually re-written into `SELECT info[0] FROM lineitem_ex ...`.